modules/product/src/main/java/com/opengamma/strata/product/credit/type/CdsConvention.java

```diff
@@ -0,0 +1,326 @@
+ */
+package com.opengamma.strata.product.credit.type;
+
+import java.time.LocalDate;
+
+import org.joda.convert.FromString;
+import org.joda.convert.ToString;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.StandardId;
+import com.opengamma.strata.basics.currency.AdjustablePayment;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.date.DaysAdjustment;
+import com.opengamma.strata.basics.date.Tenor;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.named.ExtendedEnum;
+import com.opengamma.strata.collect.named.Named;
+import com.opengamma.strata.product.TradeConvention;
+import com.opengamma.strata.product.TradeInfo;
+import com.opengamma.strata.product.common.BuySell;
+import com.opengamma.strata.product.credit.CdsImmDateLogic;
+import com.opengamma.strata.product.credit.CdsTrade;
+
+/**
+ * A market convention for credit default swap trades.
+ */
+public interface CdsConvention
+    extends TradeConvention, Named {
+
+  /**
+   * Obtains an instance from the specified unique name.
+   * 
+   * @param uniqueName  the unique name
+   * @return the convention
+   * @throws IllegalArgumentException if the name is not known
+   */
+  @FromString
+  public static CdsConvention of(String uniqueName) {
+    ArgChecker.notNull(uniqueName, "uniqueName");
+    return extendedEnum().lookup(uniqueName);
+  }
+
+  /**
+   * Gets the extended enum helper.
+   * <p>
+   * This helper allows instances of the convention to be looked up.
+   * It also provides the complete set of available instances.
+   * 
+   * @return the extended enum helper
+   */
+  public static ExtendedEnum<CdsConvention> extendedEnum() {
+    return CdsConventions.ENUM_LOOKUP;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Get the number of days between valuation date and settlement date.
+   * <p>
+   * It is usually 3 business days for standardised CDS contracts.
+   * 
+   * @return days adjustment
+   */
+  public abstract DaysAdjustment getSettlementDateOffset();
+
+  /**
+   * Get the currency of the CDS.
+   * <p>
+   * The amounts of the notional are expressed in terms of this currency.
+   * 
+   * @return the currency
+   */
+  public abstract Currency getCurrency();
+
+  //-------------------------------------------------------------------------
+  /**
+   * Creates a CDS trade based on the trade date and the IMM date logic. 
+   * <p>
+   * The start date and end date are computed from trade date with the standard semi-annual roll convention.
+   * 
+   * @param legalEntityId  the legal entity ID
+   * @param tradeDate  the trade date
+   * @param tenor  the tenor
+   * @param buySell  buy or sell
+   * @param notional  the notional
+   * @param fixedRate  the fixed rate
+   * @param refData  the reference data
+   * @return the CDS trade
+   */
+  public default CdsTrade toTrade(
```

Comment:
All methods taking `ReferenceData` should be named `createTrade()`, as per other conventions.